Index: work/src/edu/arizona/kfs/fp/businessobject/ErrorCertification.java
===================================================================
--- work/src/edu/arizona/kfs/fp/businessobject/ErrorCertification.java	(revision 0)
+++ work/src/edu/arizona/kfs/fp/businessobject/ErrorCertification.java	(revision 3011)
@@ -0,0 +1,121 @@
+/*
+ * Copyright 2009 The Kuali Foundation.
+ * 
+ * Licensed under the Educational Community License, Version 1.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * 
+ * http://www.opensource.org/licenses/ecl1.php
+ * 
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package edu.arizona.kfs.fp.businessobject;
+
+import java.util.LinkedHashMap;
+
+import org.kuali.rice.kns.bo.PersistableBusinessObjectBase;
+
+public class ErrorCertification extends PersistableBusinessObjectBase {
+    private Integer errorCertID;
+    private String expenditureDescription;
+    private String expenditureProjectBenefit;
+    private String errorDescription;
+    private String errorCorrectionReason;
+    
+    /**
+     * Gets the errorCertID attribute. 
+     * @return Returns the errorCertID.
+     */
+    public Integer getErrorCertID() {
+        return errorCertID;
+    }
+
+    /**
+     * Sets the errorCertID attribute value.
+     * @param errorCertID The errorCertID to set.
+     */
+    public void setErrorCertID(Integer errorCertID) {
+        this.errorCertID = errorCertID;
+    }
+
+    /**
+     * Gets the expenditureDescription attribute. 
+     * @return Returns the expenditureDescription.
+     */
+    public String getExpenditureDescription() {
+        return expenditureDescription;
+    }
+
+    /**
+     * Sets the expenditureDescription attribute value.
+     * @param expenditureDescription The expenditureDescription to set.
+     */
+    public void setExpenditureDescription(String expenditureDescription) {
+        this.expenditureDescription = expenditureDescription;
+    }
+
+    /**
+     * Gets the expenditureProjectBenefit attribute. 
+     * @return Returns the expenditureProjectBenefit.
+     */
+    public String getExpenditureProjectBenefit() {
+        return expenditureProjectBenefit;
+    }
+
+    /**
+     * Sets the expenditureProjectBenefit attribute value.
+     * @param expenditureProjectBenefit The expenditureProjectBenefit to set.
+     */
+    public void setExpenditureProjectBenefit(String expenditureProjectBenefit) {
+        this.expenditureProjectBenefit = expenditureProjectBenefit;
+    }
+
+    /**
+     * Gets the errorDescription attribute. 
+     * @return Returns the errorDescription.
+     */
+    public String getErrorDescription() {
+        return errorDescription;
+    }
+
+    /**
+     * Sets the errorDescription attribute value.
+     * @param errorDescription The errorDescription to set.
+     */
+    public void setErrorDescription(String errorDescription) {
+        this.errorDescription = errorDescription;
+    }
+
+    /**
+     * Gets the errorCorrectionReason attribute. 
+     * @return Returns the errorCorrectionReason.
+     */
+    public String getErrorCorrectionReason() {
+        return errorCorrectionReason;
+    }
+
+    /**
+     * Sets the errorCorrectionReason attribute value.
+     * @param errorCorrectionReason The errorCorrectionReason to set.
+     */
+    public void setErrorCorrectionReason(String errorCorrectionReason) {
+        this.errorCorrectionReason = errorCorrectionReason;
+    }
+
+    @SuppressWarnings("unchecked")
+    protected LinkedHashMap toStringMapper() {
+        LinkedHashMap m = new LinkedHashMap();
+
+        m.put("errorCertID", getErrorCertID());
+        m.put("expenditureDescription", getExpenditureDescription());
+        m.put("expenditureProjectBenefit", getExpenditureProjectBenefit());
+        m.put("errorDescription", getErrorDescription());
+        m.put("errorCorrectionReason", getErrorCorrectionReason());
+        
+        return m;
+    }
+}
