Index: work/web-root/jsp/fp/GeneralErrorCorrection.jsp
===================================================================
--- work/web-root/jsp/fp/GeneralErrorCorrection.jsp	(revision 3021)
+++ work/web-root/jsp/fp/GeneralErrorCorrection.jsp	(revision 3022)
@@ -32,6 +32,9 @@
 	<c:set var="readOnly" value="${!KualiForm.documentActions[Constants.KUALI_ACTION_CAN_EDIT]}" />
 	<fp:capitalAssetEditTab readOnly="${readOnly}"/>	
 
+    <fp:errorCertification
+        documentAttributes="${DataDictionary.ErrorCertification.attributes}" />
+
 	<gl:generalLedgerPendingEntries />
 
 	<kul:notes />
