Index: work/web-root/WEB-INF/tags/fp/errorCertification.tag
===================================================================
--- work/web-root/WEB-INF/tags/fp/errorCertification.tag	(revision 3146)
+++ work/web-root/WEB-INF/tags/fp/errorCertification.tag	(revision 3147)
@@ -23,45 +23,45 @@
 
 <c:set var="fullEntryMode" value="${KualiForm.documentActions[Constants.KUALI_ACTION_CAN_EDIT]}" />
     	
-<kul:tab tabTitle="Error Certification" defaultOpen="true" tabErrorKey="error.errorcert.*">
+<kul:tab tabTitle="Error Certification" defaultOpen="true" tabErrorKey="document.errorCertification">
      <div class="tab-container" align="center">
         <c:set var="attributes" value="${DataDictionary.ErrorCertification.attributes}" />	
      		<h3>Error Certification</h3>
         <table cellpadding="0" cellspacing="0" class="datatable" summary="Error Certification Section">
             <tr>
-                <th align="right" valign="middle" class="bord-l-b">
+                <th align="right" valign="middle" class="bord-l-b" width="50%">
                    <div align="right"><kul:htmlAttributeLabel attributeEntry="${documentAttributes.expenditureDescription}" /></div>
                 </th>
-                <td align="left" valign="middle" class="datacell">
+                <td align="left" valign="middle" class="datacell" width="50%">
                      <kul:htmlControlAttribute 
                     	attributeEntry="${documentAttributes.expenditureDescription}" property="document.errorCertification.expenditureDescription" 
                     	readOnly="${not fullEntryMode}" />
                 </td>
             </tr>
             <tr>            
-                <th align="right" valign="middle" class="bord-l-b">
+                <th align="right" valign="middle" class="bord-l-b" width="50%">
                    <div align="right"><kul:htmlAttributeLabel attributeEntry="${documentAttributes.expenditureProjectBenefit}" /></div>
                 </th>
-                <td align="left" valign="middle" class="datacell">
+                <td align="left" valign="middle" class="datacell" width="50%">
                      <kul:htmlControlAttribute 
                     	attributeEntry="${documentAttributes.expenditureProjectBenefit}" property="document.errorCertification.expenditureProjectBenefit" 
                     	readOnly="${not fullEntryMode}" />
                 </td>
             </tr>
-                <th align="right" valign="middle" class="bord-l-b">
+                <th align="right" valign="middle" class="bord-l-b" width="50%">
                    <div align="right"><kul:htmlAttributeLabel attributeEntry="${documentAttributes.errorDescription}" /></div>
                 </th>
-                <td align="left" valign="middle" class="datacell">
+                <td align="left" valign="middle" class="datacell" width="50%">
                      <kul:htmlControlAttribute 
                     	attributeEntry="${documentAttributes.errorDescription}" property="document.errorCertification.errorDescription" 
                     	readOnly="${not fullEntryMode}" />
                 </td>
             </tr>
             <tr>
-                <th align="right" valign="middle" class="bord-l-b">
+                <th align="right" valign="middle" class="bord-l-b" width="50%">
                    <div align="right"><kul:htmlAttributeLabel attributeEntry="${documentAttributes.errorCorrectionReason}" /></div>
                 </th>
-                <td align="left" valign="middle" class="datacell">
+                <td align="left" valign="middle" class="datacell" width="50%">
                      <kul:htmlControlAttribute 
                     	attributeEntry="${documentAttributes.errorCorrectionReason}" property="document.errorCertification.errorCorrectionReason" 
                     	readOnly="${not fullEntryMode}" />
