Index: work/web-root/jsp/fp/YearEndGeneralErrorCorrection.jsp
===================================================================
--- work/web-root/jsp/fp/YearEndGeneralErrorCorrection.jsp	(revision 6862)
+++ work/web-root/jsp/fp/YearEndGeneralErrorCorrection.jsp	(revision 6863)
@@ -31,6 +31,9 @@
 
 	<c:set var="readOnly" value="${!KualiForm.documentActions[Constants.KUALI_ACTION_CAN_EDIT]}" />
 	<fp:capitalAssetEditTab readOnly="${readOnly}"/>
+
+    <fp:errorCertification
+        documentAttributes="${DataDictionary.ErrorCertification.attributes}" />
 	
 	<gl:generalLedgerPendingEntries />
 
